src/services/mentees.js

```diff
@@ -779,4 +780,207 @@ module.exports = class MenteesHelper {
 			return error
 		}
 	}
+
+	/**
+	 * Get entities and organization filter
+	 * @method
+	 * @name getFilterList
+	 * @param {String} tokenInformation - token information
+	 * @param {Boolean} queryParams - queryParams
+	 * @returns {JSON} - Filter list.
+	 */
+	static async getFilterList(queryParams, tokenInformation) {
+		try {
+			let result = {
+				organizations: [],
+				entity_types: {},
+			}
+
+			let organization_ids = []
+			const organizations = await this.getOrganizationIdBasedOnPolicy(tokenInformation.id)
+			if (organizations.success && organizations.result.length > 0) {
+				organization_ids = [...organizations.result]
+
+				const defaultOrgId = await getDefaultOrgId()
+				if (defaultOrgId && !organization_ids.includes(defaultOrgId)) {
+					organization_ids.push(defaultOrgId)
+				}
+
+				let findOrganizationAndEntities = false
+				if (organization_ids.length > 0) {
+					findOrganizationAndEntities = true
+				}
+
+				if (findOrganizationAndEntities) {
+					//get organization list
+					const organizations = await userRequests.listOrganization(organization_ids)
+					if (
+						organizations &&
+						organizations.result &&
+						organizations.result &&
+						organizations.result.length > 0
+					) {
+						result.organizations = organizations.result
+					}
+
+					//get entity type with entities list
+					const getEntityTypesWithEntities = await this.getEntityTypeWithEntitiesBasedOnOrg(
+						organization_ids,
+						queryParams
+					)
+					if (getEntityTypesWithEntities.success && getEntityTypesWithEntities.result) {
+						let entityTypesWithEntities = getEntityTypesWithEntities.result
+						if (entityTypesWithEntities.length > 0) {
+							let convertedData = convertEntitiesForFilter(entityTypesWithEntities)
+							result.entity_types = filterEntitiesBasedOnParent(convertedData, defaultOrgId)
+						}
+					}
+				}
+			}
+
+			return common.successResponse({
+				statusCode: httpStatusCode.ok,
+				message: 'FILTER_FETCHED_SUCCESSFULLY',
+				result,
+			})
+		} catch (error) {
+			return error
+		}
+	}
+
+	static async getOrganizationIdBasedOnPolicy(userId) {
+		try {
+			let organization_ids = []
+
+			const menteeExtension = await menteeQueries.getMenteeExtension(userId, [
+				'external_mentor_visibility',
+				'organization_id',
+			])
+
+			if (menteeExtension.external_mentor_visibility === common.CURRENT) {
+				organization_ids.push(menteeExtension.organization_id)
+			} else if (menteeExtension.external_mentor_visibility === common.ASSOCIATED) {
+				let userOrgDetails = await userRequests.fetchDefaultOrgDetails(menteeExtension.organization_id)
```

Comment:
Though functionally this is fine, this function should be made generic since here we are not fetching the default organisation details. Leave unchanged for now.